### DISCUSSION TITLE
Issue with Integrating LuxTrust Certificate with Cypress

### DISCUSSION THREAD

**Discussion by alaeddineaboud** [cypress-io/cypress#26889] [Questions and Help]
Hello,

I am experiencing an issue with using my LuxTrust certificate while running automated tests using Cypress.

Here are the details of my issue:

The error message I am receiving is: "Certificate not found. Please ensure your device is properly connected and refresh selection."

![2023_05_31_13_50_47_Window](https://github.com/cypress-io/cypress/assets/133748882/c71031a8-64a1-4284-83b5-7178fb547f5f)

I am using version 12.12.0 of Cypress.
I have ensured that my LuxTrust device is properly connected.
I have tried clearing cache and restarting my machine, but the issue persists.
My LuxTrust certificate is valid and has not expired.
I am seeking a resolution to this issue. I appreciate any assistance or guidance you can provide to help me resolve this matter.

Best Regards,

**Comment by portyu9** [comment]
Hi @alaeddineaboud 

Cypress does support client-certificate authentication, but the supported mechanism is to configure the certificate files explicitly with `clientCertificates`.

For example, with a PFX certificate:

```ts
import { defineConfig } from 'cypress'

export default defineConfig({
  clientCertificates: [
    {
      url: 'https://your-host.example.com',
      certs: [
        {
          pfx: 'certs/test-client.pfx',
          passphrase: 'certs/passphrase.txt',
        },
      ],
    },
  ],
})
```

Cypress also supports a PEM certificate plus private-key pair.

The important limitation for your LuxTrust case is that this configuration expects **certificate/key files**. If your LuxTrust private key exists only on a hardware token/device and is accessed through the operating system/browser certificate selector, Cypress does not provide a configuration option for selecting that hardware-backed certificate from the native certificate UI.

Therefore:

- if your test environment can be issued a dedicated exportable **test** certificate in PEM/PFX form, configure it with `clientCertificates`;
- if the LuxTrust key is intentionally non-exportable and available only through the hardware device, you will likely need a separate test identity/environment or leave that hardware-certificate portion outside the automated Cypress flow.

I would not export or weaken a real production private key just to make the Cypress test work.

Clearing the Cypress/browser cache or changing `chromeWebSecurity` would not address this particular problem; it is about how the client certificate/private key is supplied for TLS authentication.

Please mark this answer as accepted if it helped, thank you.
